index.bs

```diff
@@ -11302,6 +11306,147 @@ The [=remote end steps=] given |session| and |command parameters| are:
+   otherwise if the value of <code>type</code> of the |install path spec| is equal to <code>archive-path</code> perform any implementation-defined steps to create a temporary file
+   being a result of Base64 decodeing of the value of <code>value</code> filed of |install path spec| and make its path to be the value of |archive path|,
+   otherwise |archive path| is null.
+
+1. If |archive path| if not null:
+
+   1. Perform any implementation-defined steps to unpack archive into |path|.
+
+1. Let |extension files| be a result of implementation-defined steps to list all paths in |path| folder.
+
+1. Let |extension archive| be a map matching the <code>extensions.ExtensionArchive</code> production, with the <code>files</code> field set to |extension files|.
+
+1. Return [=success=] with data |extension archive|.
+
+</div>
+
+### Commands ### {#module-extensions-commands}
+
+#### The extensions.install Command #### {#command-extensions-install}
+
+The <dfn export for=commands>extensions.install</dfn> command installs an extension for the session.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+     <pre class="cddl remote-cddl">
+     extensions.Install = (
+       method: "extensions.install",
+       params: extensions.InstallParameters
+     )
+
+     extensions.InstallParameters = {
+       extension: extensions.Extension,
+     }
+     </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+      <pre class="cddl local-cddl">
+      extensions.InstallResult = (
+        extensionId: text
+      )
+      </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for extensions.install">
+The [=remote end steps=] with |session| and |command parameters| are:
+
+1. Let the |extension| be the value of the <code>extension</code> field of |command parameters|
+
+1. If installing |extension| isn't supported return [=error=] with error code [=unsupported operation=].
+
+1. Let |install path spec| be a value of <code>installPath</code> field of the |extension|.
+
+1. Let |extension archive| be the result of [=trying=] to [=expand a install path spec=] with |install path spec|.
+
+1. Let |extension files| be a value of <code>files<code> field of |extension archive|.
+
+1. If |extension files| does not include entry <code>manifest.json</code> return [=error=] with code [=invalid extension=]
+
+1. Perform any implementation-defined steps to install the extension into |session|.
+
+1. Let |extension id| be a unique identifier of the |extension| for the browser decided by the implementation-defined steps.
+
+1. Let |allow private browsing| be a value of <code>allowPrivateBrowsing</code> field of |command parameters| if present, or false otherwise.
+
+1. Perform any implementation-defined steps to enable extension to be run in private browsing based on the valu of |allow private browsing|.
+
+1. Let |result| be a [=/map=] matching the
```

Comment:
Is there a reason this is a map and we don't just return a single `InstallResult`? I can't see a case where we would ever install more than one extension.

Reply:
I believe it is a map because InstallResult is meant to be a map (object). I think it should be defined like 

```
extensions.InstallResult = {
        extensionId: text
}
```

(`{}` instead of `()`)

Reply:
@OrKoN Definition fixed. Thank you for pointing it out. 

Where can I find a specification for this syntax? Currently I'm learning from examples in this doc.

Reply:
I still seem to see `()` instead of `{}`. The CDDL spec is here https://datatracker.ietf.org/doc/rfc8610/

Reply:
I've change the wrong place. Fixed now. Thank you for the spec!